spec/unit/provider/service/systemd_spec.rb

```diff
@@ -388,30 +388,24 @@
   # Note: systemd provider does not care about hasstatus or a custom status
   # command. I just assume that it does not make sense for systemd.
   describe "#status" do
-    it "should return running if the command returns 0" do
-      provider = provider_class.new(Puppet::Type.type(:service).new(:name => 'sshd.service'))
-      expect(provider).to receive(:execute)
-        .with(['/bin/systemctl','is-active', '--', 'sshd.service'], {:failonfail => false, :override_locale => false, :squelch => false, :combine => true})
-        .and_return(Puppet::Util::Execution::ProcessOutput.new("active\n", 0))
+    it 'when called on a service that does not exist returns absent' do
+      provider = provider_class.new(Puppet::Type.type(:service).new(:name => 'doesnotexist.service'))
+      expect(provider).to receive(:exist?).and_return(false)
+      expect(provider.status).to eq(:absent)
+    end
+
+    it 'when called on a service that does exist and is running returns running' do
+      provider = provider_class.new(Puppet::Type.type(:service).new(:name => 'doesexist.service'))
+      expect(provider).to receive(:exist?).and_return(true)
+      expect(provider).to receive(:service_command).with(:status, false).and_return(Puppet::Util::Execution::ProcessOutput.new("running\n", 0))
```

Comment:
Ideally we'd never stub methods in the thing we're trying to test, otherwise, our tests may pass when they shouldn't. So I'd recommend not stubbing the `exist?` call and instead stubbing the `execute` call that determines if the service exists or not. Similar comment about the `service_command`.